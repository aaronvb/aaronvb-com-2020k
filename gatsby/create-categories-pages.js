'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___category) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (category) => {
    const categorySlug = `/category/${_.kebabCase(category.fieldValue)}`;

    createPage({
      path: categorySlug,
      component: path.resolve('./src/templates/category-template.js'),
      context: {
        category: category.fieldValue,
      }
    });
  });
};
