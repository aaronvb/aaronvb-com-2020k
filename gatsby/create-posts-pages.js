'use strict';

const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  createPage({
    path: '/',
    component: path.resolve('./src/templates/index-template.js'),
  });
};
