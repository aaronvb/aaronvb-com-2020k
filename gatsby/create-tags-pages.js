'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (tag) => {
    const tagSlug = `/tag/${_.kebabCase(tag.fieldValue)}`;

    createPage({
      path: tagSlug,
      component: path.resolve('./src/templates/tag-template.js'),
      context: {
        tag: tag.fieldValue,
      }
    });
  });
};
