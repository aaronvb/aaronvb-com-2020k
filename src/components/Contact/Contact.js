// @flow strict
import React from 'react';
import styles from './Contact.module.scss';

type Props = {
  body: string,
}

const Contact = ({ body }: Props) => (
  <div className={styles['contact']}>
    <div className={styles['contact__inner']}>
      <h4 className={styles['page__title']} href="#contact" id="contact">Contact</h4>
      <div className={styles['contact__body']} dangerouslySetInnerHTML={{ __html: body }} />
    </div>
  </div>
);

export default Contact;