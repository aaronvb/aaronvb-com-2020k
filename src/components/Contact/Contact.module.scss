@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.contact {
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    &__inner {
        padding: 0px 20px;
    }
    h4 {
        margin: 0;
        @include padding(1, 0, 1, 0)
    }
    @include margin-bottom(1);
    p:first-child {
        margin-top: 0;
    }
    a {
        color: $color-base;
        border-bottom: 1px solid $color-light-gray;
        &:hover,
        &:focus {
          color: $color-base;
          border-bottom: 1px solid $color-base;
        }
      }
}

@include breakpoint-sm {
    .contact {
        &__inner {
            @include padding(0, 0, 0, 0);
            border-top: 1px $color-gray-border solid;
            @include margin-bottom(1.5);
        }
    }
}

@include breakpoint-md {
    .contact {
        &__inner {
            @include padding(0, 0, 0,);
            border-top: 1px $color-gray-border solid;
            @include margin-bottom(1.5);
        }
    }
}
