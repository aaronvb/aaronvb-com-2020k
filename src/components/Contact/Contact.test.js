// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Contact from './Contact';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Contact', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    body: '<p>test</p>'
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Contact { ...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
