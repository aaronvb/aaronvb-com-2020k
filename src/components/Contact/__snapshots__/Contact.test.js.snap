// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<div
  className="contact"
>
  <div
    className="contact__inner"
  >
    <h4
      className="page__title"
      href="#contact"
      id="contact"
    >
      Contact
    </h4>
    <div
      className="contact__body"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>test</p>",
        }
      }
    />
  </div>
</div>
`;
